sentinelhub/download/handlers.py

```diff
@@ -33,6 +33,7 @@ def new_download_func(self, request):
 def retry_temporal_errors(download_func):
     """ Decorator function for handling server and connection errors
     """
+    # pylint: disable=inconsistent-return-statements
```

Comment:
I was thinking about that:

- Adding `return None` seemed to be a bit misleading since it will never come to this and we don't want to give an impression that this can return `None`.
- Moving `raise` statement outside the `for` loop as you suggested is problematic because `exception` variable is defined with try-except but now we would use it outside its scope. Of course, it would work but it is also not a good practice and pylint might raise a different kind of error because of it.

I think I now found the correct solution - the only way that we could come out of the for loop is when someone would set e.g. `self.config.max_download_attempts = 0`. Hence let's just raise the following:

```Python
raise DownloadFailedException('No download attempts available')
```

Reply:
👍